tests/eas/acceptance.py

```diff
@@ -85,6 +88,52 @@ def log_result(data, log_fh):
     logging.info(result_str)
     log_fh.write(result_str)
 
+@wrapt.decorator
+def experiment_test(wrapped_test, instance, args, kwargs):
+    for experiment in instance.executor.experiments:
+        tasks = experiment['wload'].tasks.keys()
+        try:
+            wrapped_test(experiment, tasks, *args, **kwargs)
+        except AssertionError as e:
+            trace_relpath = os.path.join(experiment['out_dir'], "trace.dat")
+            logging.error("Check trace file: " + os.path.abspath(trace_relpath))
+
+class EasTest(LisaTest):
+    """
+    Base class for EAS tests
+    """
+
+    set_is_big_little = True
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        conf_file = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                 cls.conf_basename)
+
+        super(EasTest, cls)._init(conf_file, *args, **kwargs)
+
+    @memoized
+    def get_multi_assert(self, experiment, task_filter=""):
+        tasks = experiment['wload'].tasks.keys()
+        return SchedMultiAssert(experiment['out_dir'],
+                                self.te.topology,
+                                [t for t in tasks if task_filter in t])
+
+    def get_start_time(self, experiment):
+        start_times_dict = self.get_multi_assert(experiment).getStartTime()
+        return min([t['starttime'] for t in start_times_dict.values()])
+
+    @experiment_test
+    def test_first_cpu(self, experiment, tasks):
```

Comment:
Good point. I like the idea of the `_test_first_cpu` method; that way we don't duplicate the actual test logic but it's explicit that each test class includes that test.